ote/src/clj/ote/integration/import/gtfs.clj

```diff
@@ -1,33 +1,120 @@
+                  (when-let [gtfs-file-type (gtfs-spec/name->keyword name)]
+                    [gtfs-file-type (gtfs-parse/parse-gtfs-file gtfs-file-type data)])))
+          (load-zip-from-url url))))
+
+(defn gtfs-file-name [operator-id ts-id]
+  (let [new-date (java.util.Date.)
+        date (.format (java.text.SimpleDateFormat. "yyyy-MM-dd") new-date)]
+    (str date "_" operator-id "_" ts-id "_gtfs.zip")))
+
+(defn gtfs-hash [file]
+  (digest/sha-256 file))
+
+(defn db-table-name [file-name]
+  (case file-name
+    "agency.txt" :gtfs/agency
+    "stops.txt" :gtfs/stop
+    "routes.txt" :gtfs/route
+    "calendar.txt" :gtfs/calendar
+    "calendar_dates.txt" :gtfs/calendar-date
+    "shapes.txt" :gtfs/shape
+    "stop_times.txt" :gtfs/stop-time
+    "trips.txt" :gtfs/trip
+    "transfers.txt" nil
+    ))
+
+(defn save-gtfs-to-db [db gtfs-file package-id]
+  (try
+    (let [file-list (read-zip (java.io.ByteArrayInputStream. gtfs-file))]
+      (doseq [f file-list
+              :let [file-data (gtfs-parse/parse-gtfs-file (gtfs-spec/name->keyword (:name f)) (:data f))
+                    db-table-name (db-table-name (:name f))]]
+        (log/debug "File: " (:name f) " PARSED.")
+
+        (cond
+          (= db-table-name :gtfs/shape) nil                 ;; save shape data to shape[] array
+          (= db-table-name :gtfs/stop-time) nil
+          :else                                             ;; Save all rows separately
+          (doseq [fk file-data]
+            (when (and db-table-name (seq fk))
+              (specql/insert! db db-table-name (assoc fk :gtfs/package-id package-id))))))
+      (log/debug "Save-gtfs-to-db - package-id: " package-id))
+    (catch Exception e
+      (.printStackTrace e)
+      (log/warn "Error in save-gtfs-to-db" e))))
+
+
+(defn download-and-store-transit-package
+  "Download gtfs (later kalkati files also) file, upload to s3, parse and store to database.
+  Requires s3 bucket config, database settings, operator-id and transport-service-id."
+  [gtfs-config db url operator-id ts-id last-import-date]
+  (let [filename (gtfs-file-name operator-id ts-id)
+        gtfs-file (load-file-from-url url last-import-date)]
+    (if (nil? gtfs-file)
+      (log/debug "Could not found anything from given url: " url)
+      (let [new-gtfs-hash (gtfs-hash gtfs-file)
+            old-gtfs-hash (:gtfs/sha256 (first (specql/fetch db :gtfs/package
+                                                             (specql/columns :gtfs/package)
+                                                             {:gtfs/transport-operator-id operator-id
+                                                              :gtfs/transport-service-id  ts-id})))]
+        ;; IF hash doesn't match, save new and upload file to s3
+        (if (or (nil? old-gtfs-hash) (not= old-gtfs-hash new-gtfs-hash))
+          (do
+            (let [package (specql/upsert! db :gtfs/package {:gtfs/sha256                new-gtfs-hash
```

Comment:
What is the update condition for upsert?
Use `insert!` for inserting new rows unconditionally.